src/main/java/com/conveyal/datatools/manager/controllers/api/DeploymentController.java

```diff
@@ -376,6 +377,51 @@ private static boolean terminateEC2InstanceForDeployment(Request req, Response r
         return true;
     }
 
+    /**
+     * Queue a new {@link DeployJob} if there are no conflicting jobs assigned to the specified server.
+     * @param deployJob new deploy job to queue
+     * @return whether the deploy job was successfully queued or not
+     */
+    public static boolean queueDeployJob(DeployJob deployJob) {
+        String serverId = deployJob.getServerId();
+        // Check that we can deploy to the specified target. (Any deploy job for the target that is presently active will
+        // cause a halt.)
+        if (deploymentJobsByServer.containsKey(serverId)) {
+            // There is a deploy job for the server. Check if it is active.
+            DeployJob conflictingDeployJob = deploymentJobsByServer.get(serverId);
+            if (conflictingDeployJob != null && !conflictingDeployJob.status.completed) {
+                // Another deploy job is actively being deployed to the server target.
+                LOG.error("New deploy job will not be queued due to active job in progress.");
+                return false;
+            }
+        }
+
+        // For any previous deployments sent to the server/router combination, set deployedTo to null because
+        // this new one will overwrite it. NOTE: deployedTo for the current deployment will only be updated after the
+        // successful completion of the deploy job.
+        FindIterable<Deployment> deploymentsWithSameTarget = Deployment.retrieveDeploymentForServerAndRouterId(
+            serverId,
+            deployJob.getDeployment().routerId
+        );
+        for (Deployment oldDeployment : deploymentsWithSameTarget) {
+            LOG.info("Setting deployment target to null id={}", oldDeployment.id);
+            Persistence.deployments.updateField(oldDeployment.id, "deployedTo", null);
+        }
+        // Finally, add deploy job to the heavy executor.
+        DataManager.heavyExecutor.execute(deployJob);
+        deploymentJobsByServer.put(serverId, deployJob);
+        return true;
+    }
+
+    public static DeployJob checkDeploymentInProgress(String deploymentId) {
+        for (DeployJob job : deploymentJobsByServer.values()) {
+            if (job.getDeploymentId().equals(deploymentId)) {
+                return job;
+            }
+        }
+        return null;
+    }
```

Comment:
This method can be removed, it is reported as not used by IntelliJ.

Reply:
Removed.